nixos/modules/services/networking/nftables.nix

```diff
@@ -142,7 +142,7 @@ in
             '' else cfg.ruleset}
           '';
           checkPhase = lib.optionalString cfg.checkRuleset ''
-            cp $out ruleset.conf
+            cp ${cfg.rulesetFile or "$out"} ruleset.conf
```

Comment:
Does this actually work?  I do not think `cfg.rulesetFile` can be accessed in this sandbox.

Reply:
I think it should, because it is `types.path` and not `types.str`. So it would be copied to the nix store at eval time.

Reply:
Two problems of the current implementation:
1. `checkPhase` will fail when `cfg.rulesetFile` is not in the nix store.
2. The value of `${cfg.rulesetFile or "$out"}` is always `cfg.rulesetFile`.